File: backend/app/schemas/report.py
from datetime import date as date_type
from decimal import Decimal

from pydantic import BaseModel


class CategoryBreakdownItem(BaseModel):
    major_category_id: int
    major_category_name: str
    amount: Decimal


class CategoryBreakdownRead(BaseModel):
    year: int
    month: int
    items: list[CategoryBreakdownItem]
    total: Decimal


class AssetTrendPoint(BaseModel):
    date: date_type
    total_balance: Decimal


class AssetTrendRead(BaseModel):
    items: list[AssetTrendPoint]

File: frontend/src/types/report.ts
export type CategoryBreakdownItem = {
  major_category_id: number;
  major_category_name: string;
  amount: string;
};

export type CategoryBreakdownResponse = {
  year: number;
  month: number;
  items: CategoryBreakdownItem[];
  total: string;
};

export type AssetTrendPoint = {
  date: string;
  total_balance: string;
};

export type AssetTrendResponse = {
  items: AssetTrendPoint[];
};

export type AssetTrendPeriod = '3m' | '6m' | '1y' | 'all';
